test/ng/compileSpec.js

```diff
@@ -12111,6 +12111,620 @@ describe('$compile', function() {
+            $rootScope.testUrl = 'http://example.com/image.mp4'; // `http` is whitelisted
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('http://example.com/image.mp4');
+          }));
+
+          it('should accept trusted values', inject(function($rootScope, $compile, $sce) {
+            // As a MEDIA_URL URL
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsMediaUrl('javascript:foo()');
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+
+            // As a URL
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsUrl('javascript:foo()');
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+
+            // As a RESOURCE URL
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsResourceUrl('javascript:foo()');
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+          }));
+        });
+      });
+    }
+
+    describe('img[src] sanitization', function() {
+
+      it('should accept trusted values', inject(function($rootScope, $compile, $sce) {
+        element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+        // Some browsers complain if you try to write `javascript:` into an `img[src]`
+        // So for the test use something different
+        $rootScope.testUrl = $sce.trustAsMediaUrl('someuntrustedthing:foo();');
+        $rootScope.$digest();
+        expect(element.prop('src')).toEqual('someuntrustedthing:foo();');
+      }));
+
+      it('should sanitize concatenated values even if they are trusted', inject(function($rootScope, $compile, $sce) {
+        element = $compile('<img ng-prop-src="testUrl + \'ponies\'"></img>')($rootScope);
+        $rootScope.testUrl = $sce.trustAsUrl('untrusted:foo();');
+        $rootScope.$digest();
+        expect(element.prop('src')).toEqual('unsafe:untrusted:foo();ponies');
+
+        element = $compile('<img ng-prop-src="\'http://\' + testUrl2"></img>')($rootScope);
+        $rootScope.testUrl2 = $sce.trustAsUrl('xyz;');
+        $rootScope.$digest();
+        expect(element.prop('src')).toEqual('http://xyz%3B/');
+
+        element = $compile('<img ng-prop-src="testUrl3 + testUrl3"></img>')($rootScope);
+        $rootScope.testUrl3 = $sce.trustAsUrl('untrusted:foo();');
+        $rootScope.$digest();
+        expect(element.prop('src')).toEqual('unsafe:untrusted:foo();untrusted:foo();');
+      }));
+
+      it('should not sanitize other properties', inject(function($compile, $rootScope) {
```

Comment:
I feel like this is obsolete, too. It doesn't make sense in ngProp context at least.